### PR TITLE
En_Gs

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: include/functions.h [hunk 1]
```diff
@@ -3986,8 +3987,8 @@ void func_801A4D00(void);
 // void func_801A4EB0(void);
 // void func_801A4EB8(void);
 // void func_801A4FD8(void);
-// void func_801A5080(void);
-// void func_801A5100(void);
+void func_801A5080(s32 arg0);
+s32 func_801A5100(void);
```
Comment (engineer124) [inline include/functions.h:3991]:
```suggestion
s32 func_801A5100(void);
```

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 2]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+}
+
+f32 func_80998334(EnGs* this, GlobalContext* globalCtx, f32* arg2, f32* arg3, s16* arg4, f32 arg5, f32 arg6, f32 arg7,
+                  s32 arg8, s32 arg9) {
+    f32 sp2C = Math_SmoothStepToF(arg2, *arg3, arg5, arg6, arg7);
+
+    if (arg9 == 0) {
+        sp2C = Math_SmoothStepToF(arg2, *arg3, arg5, arg6, arg7);
+        this->unk_1B0[0].x = (__sinf((*arg4 % arg8) * (1.0f / arg8) * 360.0f * 0.017453292f) * *arg2) + 1.0f;
+        this->unk_1B0[0].y = 1.0f - (__sinf((*arg4 % arg8) * (1.0f / arg8) * 360.0f * 0.017453292f) * *arg2);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:361]:
```suggestion
        this->unk_1B0[0].x = (__sinf(DEGF_TO_RADF((*arg4 % arg8) * (1.0f / arg8) * 360.0f)) * *arg2) + 1.0f;
        this->unk_1B0[0].y = 1.0f - (__sinf(DEGF_TO_RADF((*arg4 % arg8) * (1.0f / arg8) * 360.0f)) * *arg2);
```
If `__sinf((*arg4 % arg8) * (1.0f / arg8) * 2 * M_PI)` works, then even better

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 3]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+}
+
+void func_809985B8(EnGs* this, GlobalContext* globalCtx) {
+    EnGs* gossipStone;
+    Vec3f sp38;
+
+    if (func_8013E2D4(&this->actor, this->unk_212, -1, 0)) {
+        Player* player = GET_PLAYER(globalCtx);
+
+        Matrix_RotateY(this->actor.shape.rot.y, 0);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:394]:
```suggestion
        Matrix_RotateY(this->actor.shape.rot.y, MTXMODE_NEW);
```

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 4]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+    }
+
+    if (this->unk_19D == 6) {
+        sp48 = Math_SmoothStepToF(&this->unk_1DC, this->unk_1E0, 0.8f, 0.1f, 0.001f);
+        sp44 = Math_SmoothStepToF(&this->unk_1E4, this->unk_1E8, 0.8f, 0.1f, 0.001f);
+        sp40 = Math_SmoothStepToF(&this->unk_1EC, this->unk_1F0, 0.8f, 0.02f, 0.001f);
+        this->unk_1B0[0].x = this->unk_1E4 + 1.0f;
+        this->unk_1B0[0].y = this->unk_1DC + 1.0f;
+        this->unk_1B0[0].x += __sinf((this->unk_1D4 % 10) * 0.1f * 360.0f * (M_PI / 180.0f)) * this->unk_1EC;
+        this->unk_1B0[0].y += __sinf((this->unk_1D4 % 10) * 0.1f * 360.0f * (M_PI / 180.0f)) * this->unk_1EC;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:737]:
```suggestion
        this->unk_1B0[0].x += __sinf(DEGF_TO_RADF((this->unk_1D4 % 10) * 0.1f * 360.0f)) * this->unk_1EC;
        this->unk_1B0[0].y += __sinf(DEGF_TO_RADF((this->unk_1D4 % 10) * 0.1f * 360.0f)) * this->unk_1EC;
```

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 5]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+                }
+            }
+        }
+
+        if (this->unk_1D4 <= 0) {
+            this->unk_19A &= ~4;
+            this->unk_21C = 3;
+            this->unk_21E = 40;
+            this->unk_1D4 = 0;
+            this->unk_19D += 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:813]:
```suggestion
            this->unk_19D++;
```

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 6]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+
+        this->unk_19E[1].y += (s16)this->unk_1DC;
+
+        if ((this->actor.world.pos.y - this->actor.home.pos.y) >= 4000.0f) {
+            this->unk_216 = 0;
+        }
+
+        if (this->actor.playerHeightRel < -12000.0f) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, 9, this->actor.world.pos.x, this->actor.world.pos.y,
+                        this->actor.world.pos.z, 0, this->actor.world.rot.y, 0, 0);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:882]:
```suggestion
            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x, this->actor.world.pos.y,
                        this->actor.world.pos.z, 0, this->actor.world.rot.y, 0, 0);
```

**Review by engineer124 — COMMENTED** [review]
Good work, some small comments

**Review by AngheloAlf — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 7]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+    this->unk_19A &= ~0x200;
+    this->actionFunc = func_80997D38;
+}
+
+void func_80997D38(EnGs* this, GlobalContext* globalCtx) {
+    static f32 D_8099A408[] = {
+        40.0f,
+        60.0f,
+        40.0f,
+        40.0f,
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:189]:
```suggestion
        40.0f
```
and reformat.

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 8]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+        case 5:
+            phi_v0 = func_80998F9C(this, globalCtx);
+            break;
+
+        case 4:
+            phi_v0 = func_809995A4(this, globalCtx);
+            break;
+    }
+
+    if (!phi_v0) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:453]: Looks like this variable isn't actually boolean, at least at this point, from what some of those functions output; does `== 0` match? Also, some of those returns are negative; can it be `s32`?

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 9]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+        sp48 = Math_SmoothStepToF(&this->unk_1DC, this->unk_1E0, 1.0f, this->unk_1E4, 0.001f);
+        this->unk_19E[0].y += (s32)(this->unk_1DC * 182.04445f);
+        if (sp48 == 0.0f) {
+            this->unk_1D4 = 0;
+            this->unk_19D = 2;
+        }
+    }
+
+    if (this->unk_19D == 2) {
+        this->unk_19E[0].y += (s32)(this->unk_1DC * 182.04445f);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:667]: 182.04445 is `0x10000 / 360.0f` (applies elsewhere too)

File: src/overlays/actors/ovl_En_Gs/z_en_gs.c [hunk 10]
```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
...
+    if (this->unk_21A > 0) {
+        this->unk_21A--;
+        if (this->unk_21A <= 0) {
+            this->unk_21A = 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/func_80997DEC.s")
+    if (!(this->unk_19A & (0x200 | 0x10)) && (this->unk_21A == 0)) {
+        if (this->collider.base.acFlags & 2) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Gs/z_en_gs.c:956]: These flags should use defines from `z64collision_check.h`: this one is `AC_HIT`.

**Review by EllipticEllipsis — COMMENTED** [review]
Good job, a few formatting/type/flag comments.

**Review by Kenix3 — APPROVED** [review]
Merge pending conflict resolution.

**Pull request merged**